src/articles/articles.guard.ts

```diff
@@ -0,0 +1,30 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { ArticlesService } from './articles.service';
+import { AuthService } from '../auth/auth.service';
+
+@Injectable()
+export class ArticlesGuard implements CanActivate {
+  constructor(
+    private authService: AuthService,
+    private articlesService: ArticlesService,
+    ) {}
+
+  canActivate(
+    context: ExecutionContext,
+  ): boolean | Promise<boolean> | Observable<boolean> {
+    const request = context.switchToHttp().getRequest();
+    const articleId = request.params.id || '';
```

Comment:
What is the purpose of setting the default value in this case? Looks like it will work without it.